packages/astro/src/runtime/server/index.ts

```diff
@@ -223,7 +225,9 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
   // This is a custom element without a renderer. Because of that, render it
   // as a string and the user is responsible for adding a script tag for the component definition.
   if (!html && typeof Component === 'string') {
-    html = await renderAstroComponent(await render`<${Component}${spreadAttributes(props)}>${children}</${Component}>`);
+    html = await renderAstroComponent(
+      await render`<${Component}${spreadAttributes(props)}${(children == null || children == '') && voidElementNames.test(Component) ? `/>` : `>${children}</${Component}>`}`
```

Comment:
Eh, `==`?

Yea, it’s that or `children === undefined || children === null || String(children) === ''`.

Reply:
I think this is good enough. I can’t think of a problem with `==` here